sources/packages/backend/libs/utilities/src/string-utils.ts

```diff
@@ -16,3 +18,127 @@ export function replaceLineBreaks(
   }
   return data.replace(REPLACE_LINE_BREAK_REGEX, options?.replaceText ?? "");
 }
+
+/**
+ * List of characters to be translated before the content is converted to an
+ * ASCII buffer. These characters do not have an equivalent character in ISO8859-1
+ * and will ended up being converted to an unexpected character.
+ */
+const ASCII_INDIRECT_CONVERSIONS: Record<string, string> = {
+  "—": "-",
+};
+
+/**
+ * Converts input into an ASCII 7 bit buffer replacing special characters
+ * by equivalent ASCII characters when possible.
+ * @param rawContent raw content in string format.
+ * @returns a string of the input with extended ASCII characters (ISO-8859-1)
+ * converted to equivalent ASCII characters. If null or undefined is provided,
+ * null will be returned.
+ */
+export function convertToASCII(rawContent?: string): string | null {
```

Comment:
What is the reason to make this method null safe?

Reply:
As it is a generic method and no longer dedicated to the SFTP only, it seems better to ensure that it can handle null values also.